pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -199,15 +193,21 @@ loop:
 
 func (s *Scheduler) scheduleBlock(block *blocks.Block) {
 	if block.SetScheduled() {
+		// deduct tokens from the token bucket according to the scheduled block's work.
+		s.tokenBucket -= float64(block.Work())
+		// check for another block ready to schedule
+		s.selectBlockToSchedule()
+		s.updateChildren(block)
 		s.events.BlockScheduled.Trigger(block)
-		fmt.Printf("Block Scheduled: %s\n", block.ID())
 	}
 }
 
 func (s *Scheduler) skipBlock(block *blocks.Block) {
 	if block.SetSkipped() && block.SetScheduled() {
+		s.selectBlockToSchedule()
+		s.updateChildren(block)
+		s.events.BlockScheduled.Trigger(block)
```

Comment:
Double trigger of incorrect event - should trigger BlockSkipped.

Reply:
I removed the double trigger, don't know how that ended up there... I also switched to BlockSkipped event and hooked that up. I was being lazy triggering BlockScheduled so I didn't have to also hook to BlockSkipped for the same events as BlockScheduled.